main.nf

```diff
@@ -347,63 +345,100 @@ workflow {
     pon_tbi = params.pon ? params.pon_index ?: BUILD_INDICES.out.pon_tbi : Channel.empty()
 
     // PREPROCESSING
-    if(!('fastqc' in skip_qc))
-        result_fastqc = FASTQC(input_sample)
-    else
-        result_fastqc = Channel.empty()
+
+    fastqc_html    = Channel.empty()
+    fastqc_version = Channel.empty()
+    fastqc_zip     = Channel.empty()
+
+    if (!('fastqc' in skip_qc)) {
+        FASTQC(input_sample)
+        fastqc_html    = FASTQC.out.html
+        fastqc_version = FASTQC.out.version
+        fastqc_zip     = FASTQC.out.zip
+    }
+
+    def bwamem2_mem_options = [:]
+
+    bwamem2_mem_options.args_bwamem2 = "-K 100000000 -M"
```

Comment:
These options can be pre-defined in a map in `conf/modules.config` like [here](https://github.com/drpatelh/chipseq/blob/7bd0e6b0e619754845e2f65bb4641671bba42bdf/conf/modules.config#L51), included via `nextflow.config` like [here](https://github.com/drpatelh/chipseq/blob/7bd0e6b0e619754845e2f65bb4641671bba42bdf/nextflow.config#L105) and then you can even append parameters like [here](https://github.com/drpatelh/chipseq/blob/7bd0e6b0e619754845e2f65bb4641671bba42bdf/main.nf#L394-L395) in the main script if required. 

Hopefully, this means the software parameters are easier to pass around the script and are more customisable by the developed/user. Also, means you dont have to initialise maps all over the place for the module settings because this is already explicitly done in `modules.config`. 

We should stick the same notation to access variable in the module files though i.e. the 5 I have needed so far are listed [here](https://github.com/drpatelh/chipseq/blob/7bd0e6b0e619754845e2f65bb4641671bba42bdf/conf/modules.config#L51-L55)

Reply:
This also means that every module will need to be passed an additional `opts` value which is a map containing the generic options for the module e.g. [here](https://github.com/drpatelh/chipseq/blob/7bd0e6b0e619754845e2f65bb4641671bba42bdf/main.nf#L410) because this will allow more flexibility to control things like output directory etc. like [here](https://github.com/drpatelh/chipseq/blob/7bd0e6b0e619754845e2f65bb4641671bba42bdf/modules/nf-core/software/picard_mergesamfiles.nf#L18)